making diacritics keys configurable

arossato opened the pull request:
Hi,

I'm proposing some changes to the code to make diacritics keys configurable. Since I prefer the ibycus way of typing polytonic Greek, and even though I know I could just edit the toogleDiacritic method, I though making the keys configurable could be a nice addition.

Probably my code is a bit redundant but I faced two problem: I'm new to writing Libreoffice extensions and I must confess I didn't find some general introduction to it (the API is instead quite well documented); then I was not able to find a way to easily debug the code: even a typo leads to the extension not being loaded without any indication on where the error is... I was not able, given my limited time, to find out how to log these errors. So I think the code I added could be cleaned up a bit, but it is working and so I'm submitting it.  

I've also updated the readme.

Please let me know.

Best regards,
andrea

arossato committed:
make diacritics keys configurable

arossato committed:
update the README file

jeremymarch commented:
Thanks for doing this.  It's a great idea.  I'll give it a closer look today or tomorrow and let you know.
All best,
Jeremy

arossato committed:
transliterate also ? -> ; and ; -> ·

jeremymarch commented:
This looks really good.  Thanks so much for doing it.  Just two small things stand out to me.  1) Could you fix the typo in "diaresisKey" – it should be spelled "diaeresisKey".  2) the size and position of the textbox widgets look different on the different platforms (linux, macOS, windows).  On macOS the boxes are very smushed.  I'm happy to fix #2 after it is merged, but I'd like to have the typo fixed before merging.  Could you make this change?

arossato committed:
fix typo in diaeresisKey

arossato commented:
I corrected the typo (sorry but I did a mistake the first time and then copied and pasted it around).

Unfortunately I do not have access to different platforms to test it so if you could fix the second problem that would be great.

Also notice that I added to characters in the transliterateLetters dictionary:   "?": ";", and  ";": "·". I find them handy.

Best,
andrea

jeremymarch commented:
It is merged.  Thanks again!
